docs/guides/transactions.md

```diff
@@ -1,6 +1,56 @@
 # Using transactions on the lakeFS file system
 
-A transaction defers file transfers and versioning operations to a queue, which is unwound sequentially on completion.
-Transactions are thread-safe and atomic, meaning that a single failure during any transaction function causes the entire transaction to abort.
+In addition to file operations, you can carry out versioning operations in your Python code using file system *transactions*.
 
-TODO: Add content
+A transaction is basically a context manager that collects all file uploads, defers them, and executes the uploads on completion of the transaction.
+They are an "all or nothing" proposition: If an error occurs during the transaction, none of the queued files are uploaded.
+
+The main features of the lakeFS file system transaction are:
+
+## Thread safety
+
+Through its use of `collections.deque` as a store for uploads, upload queueing and file transfers are thread-safe.
+
+## Atomicity
+
+If an exception occurs anywhere during the transaction, all queued file uploads are discarded:
```

Comment:
> all queued file uploads

What about the versioning ops? It's listed in the explanation of the code example below, but I believe this should be mentioned more prominently.

Reply:
Mentioned the versioning operations in [e25c6bf](https://github.com/aai-institute/lakefs-spec/pull/171/commits/e25c6bf2fc0a46929721b0dafcec76c6e02dc5b7).